modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Operation.scala

```diff
@@ -7,22 +7,23 @@ import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import enumeratum.values.{StringCirceEnum, StringEnum, StringEnumEntry}
+import eu.timepit.refined.cats._
 import eu.timepit.refined.types.numeric.NonNegLong
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object operation {
-  @derive(customizableDecoder, customizableEncoder)
+  @derive(eqv, customizableDecoder, customizableEncoder, show)
   case class Operation(
     operationIdentifier: OperationIdentifier,
     relatedOperations: Option[List[OperationIdentifier]],
     `type`: OperationType,
     status: Option[OperationStatus],
-    account: Option[AccountIdentifier],
-    amount: Option[Amount]
+    account: AccountIdentifier,
+    amount: Amount
```

Comment:
is `account` and `amount` required for every operation? if we're going to use `Operation` for creating transactions only, I'm good with that (so we're more specific). Most likely we can make `status` required as well 🤔

Reply:
I can look at this more closely next week.

Reply:
@marcinwadon Aside from _payloads_, I only see `amount` and `account` used by the _preprocess_ endpoint and I believe we can keep it required there as well. Do you see `status` used anywhere? Right now I'm not finding any usages.

Reply:
@sdavidp I don't know. You'd have to figure out how rosetta works.